lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -9,20 +10,32 @@ class PokemonOverviewPage extends StatelessWidget {
     super.key,
   });
 
-  final List<Pokemon> pokemons;
+  final Async<List<Pokemon>> pokemons;
 
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
+      scaffoldMessengerKey: scaffoldKey,
       home: Scaffold(
         appBar: AppBar(title: const Text(pokemonOverviewTitle)),
-        body: GridView.builder(
-          padding: const EdgeInsets.all(10.0),
-          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
-          itemCount: pokemons.length,
-          itemBuilder: (_, index) {
-            final pokemon = pokemons[index];
-            return PokemonTileCard(pokemon: pokemon);
+        body: pokemons.when(
+          (data) => GridView.builder(
+            padding: const EdgeInsets.all(10.0),
+            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
+            itemCount: data.length,
+            itemBuilder: (_, index) {
+              final pokemon = data[index];
+              return PokemonTileCard(pokemon: pokemon);
+            },
+          ),
+          loading: () => const Center(child: CircularProgressIndicator()),
+          error: (errorMessage) {
+            final SnackBar snackBar = SnackBar(
```

Comment:
kindly wrap this snackbar implementation in a separate function and name it with _showErrorMessageSnackbar() 🙂